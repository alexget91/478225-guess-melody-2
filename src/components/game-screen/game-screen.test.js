import React from "react";
import renderer from "react-test-renderer";
import GenreQuestionScreen from "../genre-question-screen/genre-question-screen";
import GameScreen from "./game-screen";
import ArtistQuestionScreen from "../artist-question-screen/artist-question-screen";

describe(`Game screen correctly renders after relaunch`, () => {
  it(`Genre question`, () => {
    const tree = renderer
      .create(<GameScreen type={`genre`} mistakes={1} time={0} onTimeChange={jest.fn()}>
        <GenreQuestionScreen
          screenIndex={0}
          question={{
            type: `genre`,
            genre: ``,
            answers: [{
              genre: ``,
              src: ``,
            }],
          }}
          userAnswer={[]}
          renderPlayer={jest.fn()}
          onAnswerChange={jest.fn()}
          onAnswerSubmit={jest.fn()}
        />
      </GameScreen>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Artist question`, () => {
    const tree = renderer
      .create(<GameScreen type={`artist`} mistakes={1} time={0} onTimeChange={jest.fn()}>
        <ArtistQuestionScreen
          screenIndex={0}
          question={{
            type: `artist`,
            song: {
              artist: ``,
              src: ``,
            },
            answers: [{
              artist: ``,
              picture: ``,
            }],
          }}
          onAnswer={jest.fn()}
          renderPlayer={jest.fn()}
        />
      </GameScreen>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
