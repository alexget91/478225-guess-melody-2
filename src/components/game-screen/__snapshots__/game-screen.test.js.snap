// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game screen correctly renders after relaunch Artist question 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      />
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden js-answer-input"
          data-index={0}
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="answer-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt=""
            className="artist__picture"
            src=""
          />
          
        </label>
      </div>
    </form>
  </section>
</section>
`;

exports[`Game screen correctly renders after relaunch Genre question 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      
       треки
    </h2>
    <form
      className="game__tracks js-answer-form"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            data-index={0}
            id="answer-0"
            name="answer"
            onChange={[MockFunction]}
            type="checkbox"
            value="answer-0"
          />
          <label
            className="game__check"
            htmlFor="answer-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
